test/cluster/dynamic_rackid_test.go

```diff
@@ -0,0 +1,420 @@
+								Expect(err).ToNot(HaveOccurred())
+
+								confs := strings.Split(info["namespace/test"], ";")
+								for _, conf := range confs {
+									if strings.Contains(conf, "rack-id") {
+										keyValue := strings.Split(conf, "=")
+										Expect(keyValue[1]).To(Equal("2"))
+									}
+								}
+
+								By("Updating cluster by disabling dynamic rack")
+								oldPodIDs, err = getPodIDs(ctx, aeroCluster)
+								Expect(err).ToNot(HaveOccurred())
+
+								aeroCluster.Spec.RackConfig.RackIDSource = nil
+								Expect(updateCluster(k8sClient, ctx, aeroCluster)).ToNot(HaveOccurred())
+
+								operationTypeMap = map[string]asdbv1.OperationKind{
+									aeroCluster.Name + "-1-0": asdbv1.OperationWarmRestart,
+									aeroCluster.Name + "-1-1": asdbv1.OperationWarmRestart,
+								}
+
+								aeroCluster, err = getCluster(k8sClient, ctx, clusterNamespacedName)
+								Expect(err).ToNot(HaveOccurred())
+
+								err = validateOperationTypes(ctx, aeroCluster, oldPodIDs, operationTypeMap)
+								Expect(err).ToNot(HaveOccurred())
+
+								pod = aeroCluster.Status.Pods[aeroCluster.Name+"-1-0"]
+
+								info, err = requestInfoFromNode(logger, k8sClient, ctx, clusterNamespacedName, "namespace/test", &pod)
+								Expect(err).ToNot(HaveOccurred())
+
+								confs = strings.Split(info["namespace/test"], ";")
+								for _, conf := range confs {
+									if strings.Contains(conf, "rack-id") {
+										keyValue := strings.Split(conf, "=")
+										Expect(keyValue[1]).To(Equal("1"))
+									}
+								}
+
+								By("Updating cluster by enabling dynamic rack")
+								oldPodIDs, err = getPodIDs(ctx, aeroCluster)
+								Expect(err).ToNot(HaveOccurred())
+
+								aeroCluster.Spec.RackConfig.RackIDSource = &asdbv1.RackIDSource{FilePath: aerospikePath + "/rackid"}
+								Expect(updateCluster(k8sClient, ctx, aeroCluster)).ToNot(HaveOccurred())
+
+								operationTypeMap = map[string]asdbv1.OperationKind{
+									aeroCluster.Name + "-1-0": asdbv1.OperationWarmRestart,
+									aeroCluster.Name + "-1-1": asdbv1.OperationWarmRestart,
+								}
+
+								aeroCluster, err = getCluster(k8sClient, ctx, clusterNamespacedName)
+								Expect(err).ToNot(HaveOccurred())
+
+								err = validateOperationTypes(ctx, aeroCluster, oldPodIDs, operationTypeMap)
+								Expect(err).ToNot(HaveOccurred())
+
+								pod = aeroCluster.Status.Pods[aeroCluster.Name+"-1-0"]
+
+								info, err = requestInfoFromNode(logger, k8sClient, ctx, clusterNamespacedName, "namespace/test", &pod)
+								Expect(err).ToNot(HaveOccurred())
+
+								confs = strings.Split(info["namespace/test"], ";")
+								for _, conf := range confs {
+									if strings.Contains(conf, "rack-id") {
+										keyValue := strings.Split(conf, "=")
+										Expect(keyValue[1]).To(Equal("2"))
+									}
+								}
+							},
+						)
+					},
+				)
+			},
+		)
+		Context(
+			"When doing invalid operations", func() {
```

Comment:
Can we check the error string here just to confirm the error reasons for different invalid cases?

Reply:
There are many places where we should check err message instead of just err occurred to make sure that it is failing at expected stage.
Will add in these tests as of now.